### DISCUSSION TITLE
height of CELL_CENTERED = .TRUE. values

### DISCUSSION THREAD

**Discussion by gjaeger** [firemodels/fds#13346] [Q&A] [answered]
from an input file (with dx=dy=dz=0.25m):
```
&MESH IJK=189,316,23, XB=-158.75,-111.5,-48.0,31.0,4.0,9.75/
...
&SLCF QUANTITY='OPTICAL DENSITY', CELL_CENTERED=.FALSE., ID='E2+2.0m', PBZ=6.5/
...
&SLCF QUANTITY='OPTICAL DENSITY', CELL_CENTERED=.TRUE., ID='E2+2.0mCC', PBZ=6.375/
```
In smokeview file, the "slice files" are as follows:
``` 
SLCF     1 # STRUCTURED %+2.0m &     0   189     0   316    10    10 !      3      0
*.sf
SOOT OPTICAL DENSITY
OD_C
1/m
``` 
and 
```
SLCC     1 # STRUCTURED %+2.0mCC &     0   189     0   316    10    10 !     46      1
*.sf
SOOT OPTICAL DENSITY
OD_C
1/m
```
Both slices are at the same height (K=10 or z = 6.5 m). At which height is the "cell-centered"-value determined? At 6.375 m or 6.625 m?

What's the logic behind it?

**Comment by gforney** [comment]
the node centered slice, CELL_CENTERED=.FALSE., is drawn at the nearest
node elevation value in your case Z=6.5 .  The cell centered slice,
CELL_CENTERED=.TRUE., is drawn at the nearest cell center value in your
case Z=6.375 . I ran a simple test case and verified this is the case.  Are
you seeing something different?

&HEAD CHID='TEST' /

&MESH IJK=10,10,23, XB=-158.75,-111.5,-48.0,31.0,4.0,9.75/
&SLCF QUANTITY='TEMPERATURE', CELL_CENTERED=.FALSE., ID='E2+2.0m', PBZ=6.5/
&SLCF QUANTITY='TEMPERATURE', CELL_CENTERED=.TRUE., ID='E2+2.0mCC',
PBZ=6.375/
&TAIL /

On Wed, Aug 21, 2024 at 8:22 AM Gregor Jäger ***@***.***>
wrote:

> from an input file (with dx=dy=dz=0.25m):
>
> &MESH IJK=189,316,23, XB=-158.75,-111.5,-48.0,31.0,4.0,9.75/
> ...
> &SLCF QUANTITY='OPTICAL DENSITY', CELL_CENTERED=.FALSE., ID='E2+2.0m', PBZ=6.5/
> ...
> &SLCF QUANTITY='OPTICAL DENSITY', CELL_CENTERED=.TRUE., ID='E2+2.0mCC', PBZ=6.375/
>
> In smokeview file, the "slice files" are as follows:
>
> SLCF     1 # STRUCTURED %+2.0m &     0   189     0   316    10    10 !      3      0
> *.sf
> SOOT OPTICAL DENSITY
> OD_C
> 1/m
>
> and
>
> SLCC     1 # STRUCTURED %+2.0mCC &     0   189     0   316    10    10 !     46      1
> *.sf
> SOOT OPTICAL DENSITY
> OD_C
> 1/m
>
> Both slices are at the same height (K=10 or z = 6.5 m). At which height is
> the "cell-centered"-value determined? At 6.375 m or 6.625 m?
>
> What's the logic behind it?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/firemodels/fds/discussions/13346>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6UCRR2Z7XUQPO4QIQ27OLZSSBAHAVCNFSM6AAAAABM35DCPGVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGA3TQNZVGE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Glenn Forney

**Comment by gjaeger** [comment]
@gforney Thanks.
In smokeview, I see two different slices at 6.375 m (CELL_CENTERED=T) and at 6.5m (CELL_CENTERED=F).
![test_0000](https://github.com/user-attachments/assets/4b7433b6-9daa-47df-b173-410eb68cfc68)

In smokeview file, the "slice files" are as follows:
```
SLCF     1 # STRUCTURED %E2+2.0m &     0    10     0    10    10    10 !      1      0
 test_1_1.sf
 TEMPERATURE
 temp
 C
SLCC     1 # STRUCTURED %E2+2.0mCC &   0    10     0    10    10    10 !      2      1
 test_1_2.sf
 TEMPERATURE
 temp
 C
 ``` 
 Why is  `0    10     0    10    10    10`  the same in the smokeview file? Where does smokeview get the information regarding the height as shown?

**Reply by gforney** [reply, marked as answer]
0 10 0 10 10 10 in both cases tell smokeview the cells over which the slice file cocurs.  SLCC tells smokeview that it is a cell centered slice so to place the slice at the center of the cell.  SLCF tells smokeivew it is a node centered slice so to put the slice at the edge of the cell

**Reply by gjaeger** [reply]
Thank you, now I've understood.
